core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala

```diff
@@ -427,12 +429,17 @@ class CleanerTester(
 
     def broadcastCleaned(broadcastId: Long): Unit = {
       toBeCleanedBroadcstIds -= broadcastId
-      logInfo("Broadcast" + broadcastId + " cleaned")
+      logInfo("Broadcast " + broadcastId + " cleaned")
     }
 
     def accumCleaned(accId: Long): Unit = {
       logInfo("Cleaned accId " + accId + " cleaned")
     }
+
+    def checkpointCleaned(rddId: Long): Unit = {
+      toBeCheckpointIds -= rddId
```

Comment:
Great, that's a good explanation. I see that this makes it effectively wait longer to proceed, by which time some necessary conditions are true, like that checkpoint cleanup has happened. Thank you for that detail, and to the limits of my knowledge, this looks good.